### PR TITLE
[RFC] Allow to build shared libraries

### PR DESCRIPTION
**Pull request by goedderz**
### Scope & Purpose

The motivation for this PR is to be able to split `arangodbtests` into multiple executables, without multiplying its size. This will be addressed in [a second PR (#13714)](https://github.com/arangodb/arangodb/pull/13714). If `USE_SHARED_LIBRARIES` is set, the libraries will be built as shared libraries as far as possible.

Please note that this PR is still work in progress, and not complete.

Enterprise companion PR: https://github.com/arangodb/enterprise/pull/658

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [X] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [X] No backports required

### Testing & Verification

- [X] This change is a trivial rework / code cleanup without any test coverage.
- [ ] The behavior in this PR was *manually tested*
- [X] This change is already covered by existing tests, such as *all of them*.
- [ ] This PR adds tests that were used to verify all changes:
  - [ ] Added new C++ **Unit tests**
  - [ ] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

### CODE REVIEW

File: arangod/CMakeLists.txt [hunk 1]
```diff
@@ -92,8 +103,12 @@ target_compile_definitions(arango_iresearch
   PUBLIC "$<$<CONFIG:Debug>:IRESEARCH_DEBUG=1>"
 )
 
-set(IRESEARCH_TARGET_DEPENDENCY iresearch-static) # use static to match ArangoDB
-
+get_target_property(arango_iresearch_type arango_iresearch TYPE)
```
Comment (gnusi) [inline arangod/CMakeLists.txt:106]: shared iresearch build is broken and we're considering to drop it completely, please don't use `iresearch-shared` target

**Review by gnusi — CHANGES_REQUESTED** [review]

File: arangod/CMakeLists.txt (hunk 1 above)
Comment (goedderz) [reply arangod/CMakeLists.txt:106]: Okay thanks, will change it.

**Comment by jsteemann** [comment]
Checks:

[x] Verified it compiles and links with g++10 on Ubuntu without `USE_SHARED_LIBRARIES` set
[ ] Verified it compiles and links with g++10 on Ubuntu with `USE_SHARED_LIBRARIES` set

**Comment by goedderz** [comment]
I'm abandoning this. We have too many cyclic dependencies for this to work easily, we'd probably have to refactor our code base first.

**Pull request closed without merging**